File: 백준/Silver/10828. 스택/스택.swift
let count = Int(readLine()!)!

var stack: [Int] = []

for _ in 0..<count {
    let tmpArr: [String] = readLine()!.split(separator: " ").map {String($0)}
//    tmpArr.append(contentsOf: readLine()!.components(separatedBy: " "))
    
    switch tmpArr.first {
    case "push":
        push(Int(tmpArr.last!)!)
    case "pop":
        pop()
    case "size":
        size()
    case "empty":
        empty()
    case "top":
        top()
    default:
        break
    }
}

func push(_ i : Int) {
    stack.append(i)
}

func pop(){
    print(!stack.isEmpty ? stack.popLast()! : -1)
}

func size() {
    print(stack.count)
}

func empty(){
    print(stack.isEmpty ? 1 : 0)
}

func top(){
    print(stack.isEmpty ? -1 : stack.last!)
}

File: 백준/Silver/10828. 스택/스택.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        Deque<Integer> stack = new ArrayDeque<>();

        while (N-- > 0) {
            String[] command = br.readLine().split(" ");

            switch (command[0]) {
                case "push":
                    stack.push(Integer.parseInt(command[1])); // 스택 맨 위에 추가
                    break;
                case "pop":
                    System.out.println(stack.isEmpty() ? -1 : stack.pop()); // 맨 위 제거 및 출력
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "empty":
                    System.out.println(stack.isEmpty() ? 1 : 0);
                    break;
                case "top":
                    System.out.println(stack.isEmpty() ? -1 : stack.peek()); // 맨 위 값만 확인
                    break;
            }
        }
    }
}
